sklearn/cluster/tests/test_k_means.py

```diff
@@ -892,4 +903,81 @@ def test_less_centers_than_unique_points():
+
+
+def _sort_cluster_centers_and_labels(centers, labels):
+    sort_index = np.argsort(centers, axis=0)[:, 0]
+    sorted_labels = np.zeros_like(labels)
+    for i, l in enumerate(sort_index):
+        sorted_labels[labels == l] = i
+    return centers[sort_index, :], sorted_labels
+
+
+def test_weighted_vs_repeated():
+    # a sample weight of N should yield the same result as an N-fold
+    # repetition of the sample
+    sample_weights = np.random.randint(1, 5, size=n_samples)
+    X_repeat = np.repeat(X, sample_weights, axis=0)
+    for estimator in [KMeans(n_clusters=n_clusters, random_state=42),
+                      MiniBatchKMeans(n_clusters=n_clusters, batch_size=10,
+                                      random_state=42)]:
+        est_weighted = clone(estimator).fit(X, sample_weights=sample_weights)
+        est_repeated = clone(estimator).fit(X_repeat)
+        centers_1, labels_1 = _sort_cluster_centers_and_labels(
+                est_repeated.cluster_centers_, est_repeated.labels_)
+        centers_2, labels_2 = _sort_cluster_centers_and_labels(
+                est_weighted.cluster_centers_, np.repeat(est_weighted.labels_,
+                                                         sample_weights))
+        assert_almost_equal(v_measure_score(labels_1, labels_2), 1.0)
+        if not isinstance(estimator, MiniBatchKMeans):
+            assert_almost_equal(centers_1, centers_2)
+
+
+def test_unit_weights_vs_no_weights():
+    # not passing any sample weights should be equivalent
+    # to all weights equal to one
+    sample_weights = np.ones(n_samples)
+    for estimator in [KMeans(n_clusters=n_clusters, random_state=42),
+                      MiniBatchKMeans(n_clusters=n_clusters, random_state=42)]:
+        est_1 = clone(estimator).fit(X)
+        est_2 = clone(estimator).fit(X, sample_weights=sample_weights)
+        centers_1, labels_1 = _sort_cluster_centers_and_labels(
+                est_1.cluster_centers_, est_1.labels_)
+        centers_2, labels_2 = _sort_cluster_centers_and_labels(
+                est_2.cluster_centers_, est_2.labels_)
+        assert_almost_equal(v_measure_score(labels_1, labels_2), 1.0)
+        assert_almost_equal(centers_1, centers_2)
+
+
+def test_scaled_weights():
+    # scaling all sample weights by a common factor
+    # shouldn't change the result
+    sample_weights = np.ones(n_samples)
+    for estimator in [KMeans(n_clusters=n_clusters, random_state=42),
+                      MiniBatchKMeans(n_clusters=n_clusters, random_state=42)]:
+        est_1 = clone(estimator).fit(X)
+        est_2 = clone(estimator).fit(X, sample_weights=0.5*sample_weights)
+        centers_1, labels_1 = _sort_cluster_centers_and_labels(
+                est_1.cluster_centers_, est_1.labels_)
+        centers_2, labels_2 = _sort_cluster_centers_and_labels(
+                est_2.cluster_centers_, est_2.labels_)
+        assert_almost_equal(v_measure_score(labels_1, labels_2), 1.0)
```

Comment:
You should be able to get the perfect v_measure even without sorting. This makes for a red herring when reading the code

Reply:
You're right, the sorting of the labels was before I had discovered v_measure_score and is now redundant